plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3373,4 +3370,16 @@ public void testTargetMaxFileSize()
                 // so just to be safe we check if it is not much bigger
                 .forEach(row -> assertThat((Long) row.getField(0)).isBetween(1L, maxSize.toBytes() * 3));
     }
+
+    @Test
+    public void testDroppingIcebergAndCreatingANewTableWithTheSameNameShouldBePossible()
+    {
+        assertUpdate("CREATE TABLE test_iceberg_recreate (x1 BIGINT)");
+        assertThat(query("SELECT count(x1) FROM test_iceberg_recreate")).matches("VALUES BIGINT '0'");
```

Comment:
Would be a better test if you insert something here so that after the drop there's a different result